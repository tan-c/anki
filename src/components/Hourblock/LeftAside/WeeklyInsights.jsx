import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';

import {
  todayTotalSelector,
  thisWeekMetricsSelector,
  lastWeekMetricsSelector,
  yearlyAverageSelector,
  todayTotalComplianceSelector
} from 'utility-redux/dailyRecord';
import { calorieWeeklySelector } from 'utility-redux/dailyMeasurement';

export class WeeklyInsights extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const {
      todayTotal,
      yearlyAverage,
      thisWeekTotal,
      lastWeekTotal,
      thisWeekTotalCalorie,
      calorieWeeklyMaintain,
      calorieWeeklyWeightLoss,
    } = this.props;

    return (
      <div style={{
        marginTop: 20
      }}
      >
        <span className="text-center">
          {(thisWeekTotalCalorie / 1000).toFixed(1)}
          K/
          {(calorieWeeklyWeightLoss / 1000).toFixed(1)}
          K
        </span>
        <span className="text-center">
          {(thisWeekTotalCalorie / 1000).toFixed(1)}
          K/
          {(calorieWeeklyMaintain / 1000).toFixed(1)}
          K
        </span>

        <div style={{
          position: 'absolute',
          width: '100%',
          left: 0,
          bottom: 0,
          fontSize: 12,
          textAlign: 'center',
          height: 40
        }}
        >
          <div style={{
            left: 0,
            bottom: 0,
            position: 'absolute',
            textAlign: 'center',
            width: '100%',
          }}
          >
            {`${todayTotal}/${yearlyAverage.toFixed(1)}`}
            <br />
            {`${thisWeekTotal}/${lastWeekTotal}`}
          </div>

          <div
            className="bg-orange" style={{
              width: `${todayTotal <= yearlyAverage ? (todayTotal * 100) / yearlyAverage : 100}%`,
              left: 0,
              bottom: 0,
              height: 40,
              zIndex: -1,
              boxSizing: 'border-box',
              lineHeight: 20,
              fontSize: 10,
              position: 'relative'
            }}
          />
        </div>
      </div>
    );
  }
}

WeeklyInsights.defaultProps = {
  yearlyAverage: 0,
  todayTotal: 0,
  // todayTotalCompliance: 0,
  thisWeekTotal: 0,
  lastWeekTotal: 0,
  thisWeekTotalCalorie: 0,
  calorieWeeklyMaintain: 0,
  calorieWeeklyWeightLoss: 0,
};

WeeklyInsights.propTypes = {
  yearlyAverage: PropTypes.number,
  todayTotal: PropTypes.number,
  // todayTotalCompliance: PropTypes.number,
  thisWeekTotal: PropTypes.number,
  lastWeekTotal: PropTypes.number,
  thisWeekTotalCalorie: PropTypes.number,
  calorieWeeklyMaintain: PropTypes.number,
  calorieWeeklyWeightLoss: PropTypes.number,
};

function mapStateToProps(state, ownProps) {
  return {
    dailyMeasurements: state.dailyMeasurements,

    todayTotal: todayTotalSelector(state),
    // todayTotalCompliance: todayTotalComplianceSelector(state),
    yearlyAverage: yearlyAverageSelector(state),
    thisWeekTotal: thisWeekMetricsSelector(state)[0],

    thisWeekTotalCalorie: thisWeekMetricsSelector(state)[3],

    lastWeekTotal: lastWeekMetricsSelector(state)[0],

    calorieWeeklyMaintain: calorieWeeklySelector(state)[0],
    calorieWeeklyWeightLoss: calorieWeeklySelector(state)[1],
  };
}

// function mapDispatchToProps(dispatch) {
//   return {
//   };
// }

export default connect(mapStateToProps)(WeeklyInsights);
