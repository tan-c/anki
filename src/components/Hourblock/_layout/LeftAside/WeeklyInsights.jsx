import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';

import {
  todayTotalSelector,
  thisWeekMetricsSelector,
  lastWeekMetricsSelector,
  yearlyAverageSelector,
  todayTotalComplianceSelector
} from 'utility-redux/hourblock/dailyRecord';
import { calorieWeeklySelector } from 'utility-redux/hourblock/dailyMeasurement';

export class WeeklyInsights extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const {
      todayTotal,
      yearlyAverage,
      thisWeekTotal, lastWeekTotal,
      thisWeekTotalCalorie,
      calorieWeeklyMaintain, calorieWeeklyWeightLoss,
    } = this.props;

    return (
      <div data-role="weekly-insights">
        <span className="text-center">
          {(thisWeekTotalCalorie / 1000).toFixed(1)}
K/
          {(calorieWeeklyWeightLoss / 1000).toFixed(1)}
K
        </span>
        <span className="text-center">
          {(thisWeekTotalCalorie / 1000).toFixed(1)}
K/
          {(calorieWeeklyMaintain / 1000).toFixed(1)}
K
        </span>

        <div id="pomoBtn">
          {todayTotal}
          {' '}
/
          {yearlyAverage.toFixed(1)}
          <br />
          {thisWeekTotal}
          {' '}
/
          {lastWeekTotal}
          <div
            className="bg-orange" style={{
              width: `${todayTotal <= yearlyAverage ? (todayTotal * 100) / yearlyAverage : 100}%`, position: 'absolute', left: '0px', bottom: '0px', height: '40px', zIndex: '-1',
            }}
          />
        </div>
      </div>
    );
  }
}

WeeklyInsights.defaultProps = {
  yearlyAverage: 0,
  todayTotal: 0,
  // todayTotalCompliance: 0,
  thisWeekTotal: 0,
  lastWeekTotal: 0,
  thisWeekTotalCalorie: 0,
  calorieWeeklyMaintain: 0,
  calorieWeeklyWeightLoss: 0,
};

WeeklyInsights.propTypes = {
  yearlyAverage: PropTypes.number,
  todayTotal: PropTypes.number,
  // todayTotalCompliance: PropTypes.number,
  thisWeekTotal: PropTypes.number,
  lastWeekTotal: PropTypes.number,
  thisWeekTotalCalorie: PropTypes.number,
  calorieWeeklyMaintain: PropTypes.number,
  calorieWeeklyWeightLoss: PropTypes.number,
};

function mapStateToProps(state, ownProps) {
  return {
    dailyMeasurements: state.dailyMeasurements,

    todayTotal: todayTotalSelector(state),
    // todayTotalCompliance: todayTotalComplianceSelector(state),
    yearlyAverage: yearlyAverageSelector(state),
    thisWeekTotal: thisWeekMetricsSelector(state)[0],

    thisWeekTotalCalorie: thisWeekMetricsSelector(state)[3],

    lastWeekTotal: lastWeekMetricsSelector(state)[0],

    calorieWeeklyMaintain: calorieWeeklySelector(state)[0],
    calorieWeeklyWeightLoss: calorieWeeklySelector(state)[1],
  };
}

// function mapDispatchToProps(dispatch) {
//   return {
//   };
// }

export default connect(mapStateToProps)(WeeklyInsights);
