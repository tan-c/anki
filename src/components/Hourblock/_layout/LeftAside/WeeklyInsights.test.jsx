import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { WeeklyInsights } from './WeeklyInsights';

function setup() {
  const props = {
  };

  return shallow(<WeeklyInsights {...props} />);
}

describe('WeeklyInsights Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("div[data-role='weekly-insights']")).toHaveLength(1);
    });
  });
});
