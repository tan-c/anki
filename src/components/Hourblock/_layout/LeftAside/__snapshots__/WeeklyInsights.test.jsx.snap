// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyInsights Unit Tests when normally set up renders correctly 1`] = `
<div
  data-role="weekly-insights"
>
  <span
    className="text-center"
  >
    0.0
    K/
    0.0
    K
  </span>
  <span
    className="text-center"
  >
    0.0
    K/
    0.0
    K
  </span>
  <div
    id="pomoBtn"
  >
    0
     
    /
    0.0
    <br />
    0
     
    /
    0
    <div
      className="bg-orange"
      style={
        Object {
          "bottom": "0px",
          "height": "40px",
          "left": "0px",
          "position": "absolute",
          "width": "NaN%",
          "zIndex": "-1",
        }
      }
    />
  </div>
</div>
`;
